Homework7.py

```diff
@@ -0,0 +1,120 @@
+class GameObject(object):
+    class_name = ""
+    desc = ""
+    objects = {}
+
+    def __init__(self, name):
+        self.name = name
+        GameObject.objects[self.class_name] = self
+
+    def get_desc(self):
+        return self.class_name + "\n" + self.desc
+
+
+class Rogue(GameObject):
+    def __init__(self, name):
+        self.class_name = "rogue"
+        self.health = 3
+        self.desc = "A little rogue"
+        super().__init__(name)
+
+    @property
+    def desc(self):
+        if self.health >= 3:
+            return self._desc
+        elif self.health == 2:
+            health_line = "It has a wound on its knee."
+        elif self.health == 1:
+            health_line = "Its left arm has been cut off!"
+        else:
+            health_line = "It is dead"
+        return self._desc + "\n" + health_line
+
+    @desc.setter
+    def desc(self, value):
+        self._desc = value
+
+
+class Goblin(GameObject):
+    def __init__(self, name):
+        self.class_name = "goblin"
+        self.health = 3
+        self.desc = "A foul creature"
+        super().__init__(name)
+
+    @property
+    def desc(self):
+        if self.health >= 3:
+            return self._desc
+        elif self.health == 2:
+            health_line = "It has a wound on its knee."
+        elif self.health == 1:
+            health_line = "Its left arm has been cut off!"
+        else:
+            health_line = "It is dead"
+        return self._desc + "\n" + health_line
+
+    @desc.setter
+    def desc(self, value):
+        self._desc = value
+
+
+def hit(noun):
```

Comment:
why it's separate functions not methods? it would be good to create character class